gen/generator.go

```diff
@@ -907,7 +917,8 @@ type Property struct {
 	DefaultValue       string
 	Versions           string
 	NamedPropertyClass string
-	ValidValues        []string
+	ValidValues        []interface{}
+	ValidValuesNames   []string
```

Comment:
Why do we do change the ValidValues variable and then add one called ValidValuesNames, is it not easier to introduce a new variable of type []interface called ValidValuesMap?

Also we do not add the number equivalent now to the ValidValues (in your case ValidValuesNames), is this something we should?